pennylane/devices/qubit/apply_operation.py

```diff
@@ -198,6 +199,29 @@ def _(op: type_op, state):
     return _apply_operation_default(op, state, is_state_batched, debugger)
 
 
+@apply_operation.register
+def apply_mid_measure(op: MidMeasureMP, state, is_state_batched: bool = False, debugger=None):
```

Comment:
If we are going to change the interface for `apply_operation`, we should make sure we are doing it for *all* `apply_operation`.  I'd like that function to be consistent with the return shape. 

Our other alternative is to have a special different function for applying mid-circuit meausurements and not supporting it with the `apply_operation` function. This would have the benefit that we wouldn't have to make a breaking change to `apply_operation`.

Reply:
Make sense, no need to break `apply_operation` right now.